src/driver/main.cpp

```diff
@@ -565,11 +568,42 @@ struct verify : command<verify>
         auto t = c.ct.get_target();
         auto m = c.parameters.generate(p, t, true, c.l.batch);
 
+
+        //TODO remove this and make the driver able to figure out datatype most used in the model
+        // then set the tolerances appropriately. Need to check here because c.to_fp16 only set
+        // after argument_parser.parse() is run. This code is complicated because there's not a good way
+        // to change the default tolerances after reading `--fp16` but before reading `--rms-tol`, `--atol`,
+        // and `--rtol`.
+        migraphx::verify::tolerance tols{};
+        if(c.to_fp16)
+        {
+            tols = migraphx::verify::tolerance{8e-2, 4e-2, 4e-2};
+        }
+        auto dbl_comp = [&](double a, double b){
+            return (std::abs(a - b) <= std::numeric_limits<double>::epsilon());
+        };
+        if(not dbl_comp(this->rms_tol, -1.))
```

Comment:
You can probably use `float_equal` instead. It should work for `double` as well.